pkg/arvo/app/btc-wallet.hoon

```diff
@@ -704,102 +690,148 @@
+  =?  blocks  (gth gap 50)  ~
+  ::
+  =/  cards=(list card)
     ;:  weld
-::      (retry-addrs network)
       retry-ahistorical-txs
       (retry-pend-piym network)
+      (retry-block-info blocks)
     ==
-  ::
-  ++  retry-ahistorical-txs
-    ^-  (list card)
-    %+  turn  ~(tap in ahistorical-txs)
-    |=  =txid
-    (poke-provider [%tx-info txid])
-
-  ::
-  ++  retry-scans
-    |=  =network
-    ^-  (list card)
-    %-  zing
-    %+  murn  ~(tap by scans)
-    |=  [[=xpub:bc =chyg] =batch]
-    ?.  =(network network:(~(got by walts) xpub))  ~
-    `-:(req-scan batch xpub chyg)
-  ::  +retry-addrs: get info on addresses with unconfirmed UTXOs
-  ::
-  ++  retry-addrs
-    |=  =network
-    ^-  (list card)
-    %-  zing
-    %+  murn  ~(val by walts)
-    |=  w=walt
-    ?.  =(network network.w)  ~
-    ^-  (unit (list card))
-    :-  ~
-    %+  turn  ~(tap by wach.w)
-    |=  [a=address *]
-    (poke-provider [%address-info a])
-  ::  +retry-txs: get info on txs without enough confirmations
-  ::
-  ++  retry-txs
-    |=  =network
-    ^-  (list card)
-    %+  murn  ~(tap by history)
-    |=  [=txid =hest]
-    =/  w  (~(get by walts) xpub.hest)
-    ?~  w  ~
-    ?.  =(network network.u.w)  ~
-    ?:  (gte confs.hest confs.u.w)  ~
-    `(poke-provider [%tx-info txid])
-  ::
-  ++  retry-poym
-    |=  =network
-    ^-  (list card)
-    ?~  txbu.poym  ~
-    =/  w  (~(get by walts) xpub.u.txbu.poym)
-    ?~  w  ~
-    ?.  =(network network.u.w)  ~
-    %+  weld
-      ?~  signed-tx.u.txbu.poym  ~
-      ~[(poke-provider [%broadcast-tx u.signed-tx.u.txbu.poym])]
-    %+  turn  txis.u.txbu.poym
-    |=  =txi
-    (poke-provider [%raw-tx ~(get-txid txb:bl u.txbu.poym)])
-  ::  +retry-pend-piym: check whether txids in pend-piym are in mempool
-  ::
-  ++  retry-pend-piym
-    |=  =network
-    ^-  (list card)
-    %+  murn  ~(tap by pend.piym)
-    |=  [=txid p=payment]
-    =/  w  (~(get by walts) xpub.p)
-    ?~  w  ~
-    ?.  =(network network.u.w)  ~
-    `(poke-provider [%tx-info txid])
-  --
+  =?  cards  ?|(!connected.p (gth gap 0))
+    ;:  weld  cards
+      (retry-poym network)
+      (retry-txs network)
+      (retry-scans network)
+    ==
+  =?  cards  ?&(?=(^ blockhash) ?=(^ blockfilter) (gth gap 0))
+    ;:  weld  cards
+      (retry-filtered-addrs network u.blockhash u.blockfilter)
+    ==
+  =?  cards  (gth gap 50)
+    ;:  weld  cards
+      (retry-addrs network)
+    ==
+  :-  cards
+  %_  state
+    prov  `p(connected %.y)
+    btc-state  [block fee now.bowl]
+  ==
+::
+++  retry-block-info
+  |=  blocks=(list @ud)
+  %+  turn  blocks
+  |=  block=@ud
+  (poke-provider %block-info `block)
+::
+++  retry-ahistorical-txs
+  ^-  (list card)
+  %+  turn  ~(tap in ahistorical-txs)
+  |=  =txid
+  (poke-provider [%tx-info txid])
+::
+++  retry-scans
+  |=  =network
+  ^-  (list card)
+  %-  zing
+  %+  murn  ~(tap by scans)
+  |=  [[=xpub:bc =chyg] =batch]
+  ?.  =(network network:(~(got by walts) xpub))  ~
```

Comment:
Crashing here in the got by seems like a bad idea. Let's make sure we produce ~ instead